Training5/src/task1/src/App.js

```diff
@@ -1,44 +1,55 @@
-import { useEffect} from "react";
+import { useEffect, useState } from "react";
 import "./App.css";
 import Filter from "./components/Filter";
 import Todo from "./components/Todo";
 import TodoCreate from "./components/TodoCreate";
 import { useSelector, useDispatch } from "react-redux";
 import { fetchTodos } from "./api/index";
 import {
-  DeleteTodo,
-  GetTodos,
-  OpenFormEdit,
-  SetCompletedTodo,
+  deleteTodo,
+  getTodos,
+  openFormEdit,
+  setCompletedTodo,
 } from "./actions/todo";
 import TodoEdit from "./components/TodoEdit";
 
 function App() {
   const dispatch = useDispatch();
-  const listTodos = useSelector((state) => state.todo.renderList);
+  const listTodos = useSelector((state) => state.todo.todos);
+  const filter = useSelector((state) => state.todo.isLoadCompleted);
+  const searchValue = useSelector((state) => state.todo.searchValue);
+  const [listTodosRender, setListTodosRender] = useState(listTodos);
 
   useEffect(() => {
     fetchTodos().then((newList) => {
-      const action = GetTodos(newList);
+      const action = getTodos(newList);
       dispatch(action);
     });
   }, [dispatch]);
 
+  useEffect(() => {
```

Comment:
@phuongluongZigvy please use useMemo here, don't need useState for listTodosRender, using useState + useEffect, first render will map data is not correct